src/bodywork/stage.py

```diff
@@ -234,24 +234,24 @@ def run_stage(
     log = bodywork_log_factory()
     log.info(f'attempting to run stage={stage_name} from {repo_branch} branch of repo'
              f' at {repo_url}')
-    download_project_code_from_repo(repo_url, repo_branch, cloned_repo_dir)
-    path_to_stage_dir = cloned_repo_dir / stage_name
-    stage = stage_factory(path_to_stage_dir)
     try:
+        download_project_code_from_repo(repo_url, repo_branch, cloned_repo_dir)
+        path_to_stage_dir = cloned_repo_dir / stage_name
+        stage = stage_factory(path_to_stage_dir)
         _install_python_requirements(stage.requirements_file_path)
-        run(['python', stage.executable_script_path], check=True)
+        run(['python', stage.executable_script_path], check=True, encoding='utf-8')
         log.info(f'successfully ran stage={stage_name} from {repo_branch} branch of repo'
                  f' at {repo_url}')
-    except CalledProcessError as e:
-        stage_failure_exception = BodyworkStageFailure(stage_name, e.cmd, e.stderr)
+    except Exception as e:
+        stage_failure_exception = BodyworkStageFailure(stage_name, e)
         log.error(stage_failure_exception)
-        raise stage_failure_exception from e
+        raise stage_failure_exception
```

Comment:
Shouldn't we leave in the `from e` so that we preserve the trace-back?

Reply:
Sorry I misunderstood the Python docs when it said: 
'Exception chaining happens automatically when an exception is raised inside an except or finally section'
That's now been fixed.